armory/data/datasets.py

```diff
@@ -479,12 +479,81 @@ def german_traffic_sign(
     )
 
 
+class LibriSpeechDevCleanContext:
+    def __init__(self):
+        self.input_type = np.int64  # However, stores values in int16 range
+        self.input_min = -(2 ** 15)
+        self.input_max = 2 ** 15 - 1
+        self.x_shape = (None,)
+        self.sample_rate = 16000
+        self.quantization = 2 ** 15
+        self.output_type = np.float32
+        self.output_min = -1.0
+        self.output_max = 1.0
+
+
+librispeech_dev_clean_context = LibriSpeechDevCleanContext()
+
+
+def librispeech_dev_clean_dataset_canonical_preprocessing(batch):
+    context = librispeech_dev_clean_context
+    if batch.dtype == np.object:
+        for x in batch:
+            if x.dtype != context.input_type:
+                raise ValueError(f"input x dtype {x.dtype} != {context.input_type}")
+            assert x.max() <= context.input_max
+            assert x.min() >= context.input_min
+
+        batch = np.array(
+            [x.astype(context.output_type) / context.quantization for x in batch],
+            dtype=object,
+        )
+
+        for x in batch:
+            assert x.dtype == context.output_type
+            assert x.max() <= context.output_max
+            assert x.min() >= context.output_min
+        return batch
+
+    if batch.ndim != len(context.x_shape) + 1:
+        print(batch)
+        raise ValueError(f"input batch dim {batch.ndim} != {len(context.x_shape) + 1}")
+    if batch.dtype != context.input_type:
+        raise ValueError(f"input batch dtype {batch.dtype} != {context.input_type}")
+    assert batch.max() <= context.input_max
+    assert batch.min() >= context.input_min
+
+    batch = batch.astype(context.output_type) / context.quantization  # 2**15
+    assert batch.dtype == context.output_type
+    assert batch.max() <= context.output_max
+    assert batch.min() >= context.output_min
+
+    return batch
+
+
+def preprocessing_chain(*args):
+    """
+    Wraps and returns a sequence of functions
+    """
+    functions = [x for x in args if x is not None]
+    if not functions:
+        return None
+
+    def wrapped(x):
+        for function in functions:
+            x = function(x)
+        return x
+
+    return wrapped
+
+
 def librispeech_dev_clean(
     split_type: str = "train",
     epochs: int = 1,
     batch_size: int = 1,
     dataset_dir: str = None,
-    preprocessing_fn: Callable = None,
+    preprocessing_fn: Callable = librispeech_dev_clean_dataset_canonical_preprocessing,
```

Comment:
For other methods, we were not giving users the option to pass preprocessing_fn to the function, and hardcoding the canonical preprocessing in the call to _generator_from_tfds. We could change the other methods, but probably should be consistent across the methods.

Reply:
For training, users will likely want to have control over the preprocessing. Further, for testing, it is useful to have an easy way to pull the raw form of the data by setting preprocessing_fn to None.

Reply:
Also, I need this to be callable to get training to work. My preference would be to change the others to this sort of format. If we don't give users the ability to modify preprocessing_fn during evaluation, it shouldn't be a problem (I think)

Reply:
Sounds good. I will make a small PR updating the others.